smart-contract-verifier/smart-contract-verifier-server/src/run.rs

```diff
@@ -16,14 +12,56 @@ use smart_contract_verifier_proto::blockscout::smart_contract_verifier::v1::{
     sourcify_verifier_server::SourcifyVerifierServer, vyper_verifier_actix::route_vyper_verifier,
     vyper_verifier_server::VyperVerifierServer,
 };
-use std::{net::SocketAddr, sync::Arc};
+use startuper::StartupSettings;
+use std::sync::Arc;
 use tokio::sync::Semaphore;
 
-pub async fn run(settings: Settings) -> Result<(), anyhow::Error> {
-    init_logs(settings.jaeger);
+#[derive(Clone)]
+struct HttpRouter {
+    solidity_verifier: Option<Arc<SolidityVerifierService>>,
+    vyper_verifier: Option<Arc<VyperVerifierService>>,
+    sourcify_verifier: Option<Arc<SourcifyVerifierService>>,
+    health: Arc<HealthService>,
+}
 
-    let compilers_lock = Arc::new(Semaphore::new(settings.compilers.max_threads.get()));
+impl startuper::HttpRouter for HttpRouter {
+    fn register_routes(&self, service_config: &mut actix_web::web::ServiceConfig) {
+        let service_config =
+            service_config.configure(|config| route_health(config, self.health.clone()));
+
+        let service_config = if let Some(solidity) = &self.solidity_verifier {
+            service_config.configure(|config| route_solidity_verifier(config, solidity.clone()))
+        } else {
+            service_config
+        };
+        let service_config = if let Some(vyper) = &self.vyper_verifier {
+            service_config.configure(|config| route_vyper_verifier(config, vyper.clone()))
+        } else {
+            service_config
+        };
+        if let Some(sourcify) = &self.sourcify_verifier {
```

Comment:
i suggest making it as previous lines, so it would help refactoring later 
i.e. adding new service will be just copy paste lines, without adding `let ...` for this line etc
not forcing tho